akka-distributed-data/src/main/scala/akka/cluster/ddata/ReplicatedData.scala

```diff
@@ -77,38 +93,90 @@ trait DeltaReplicatedData extends ReplicatedData {
 }
 
 /**
+ * The delta must implement this type.
+ */
+trait ReplicatedDelta extends ReplicatedData {
+  /**
+   * The empty full state. This is used when a delta is received
+   * and no existing full state exists on the receiving side. Then
+   * the delta is merged into the `zero` to create the initial full state.
+   */
+  def zero: DeltaReplicatedData
+}
+
+/**
+ * Marker that specifies that the deltas must be applied in causal order.
+ * Note that if the full state type `T` is different from the delta type `D`
+ * it is the delta `D` that should be marked with this.
```

Comment:
Maybe add a note that the overhead with causal delivery is bigger than without it, so if possible its good to avoid?